addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InformationDisclosureSuspiciousCommentsScanRule.java

```diff
@@ -90,19 +114,13 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isText()) {
 
             if (ResourceIdentificationUtils.isJavaScript(msg)) {
-                // Just treat as text
-                String[] lines = msg.getResponseBody().toString().split("\n");
-                for (String line : lines) {
+                for (String comment : getJsComments(msg.getResponseBody().toString())) {
                     for (Pattern pattern : patterns) {
-                        Matcher m = pattern.matcher(line);
+                        Matcher m = pattern.matcher(comment);
                         if (m.find()) {
                             recordAlertSummary(
                                     alertMap,
-                                    new AlertSummary(
-                                            pattern.toString(),
-                                            line,
-                                            Alert.CONFIDENCE_LOW,
```

Comment:
I don't think we should be changing the confidence, we are still not extracting comments (e.g. `"// FROM"`), and IMO we should not be saying that we are checking comments, "likely comments".

Reply:
Okay, that's fair

Reply:
Worth adding a test.

Reply:
And have performance test.

Reply:
> Worth adding a test.

Do you mean a quoted string that seems like a comment?

Reply:
> And have performance test.

Got advice or a vision for that?

Reply:
https://github.com/zaproxy/zap-extensions/pull/5813#discussion_r1816618026

Right, for both type of comments.

https://github.com/zaproxy/zap-extensions/pull/5813#discussion_r1816618478

Similar to `PolyfillCdnScriptScanRuleUnitTest.shouldRunQuickly`. (Probably requires going through some common/big websites to find a good source first.)

Reply:
For the first one would something like: https://www.regexplanet.com/share/index.html?share=yyyyf7fc8kr be better? I'm sure there are still some edge cases not handled but it's gotta be better than the previous behavior (IMHO).
![image](https://github.com/user-attachments/assets/b6718b0b-6865-4050-bd69-368220504b80)

Reply:
What's the performance of the new one compared to the older? That's more important, the existing regex will already reduce FPs and hopefully without much or any performance degradation.

Reply:
I'll work on some live testing over the weekend.